pywemo/ssdp.py

```diff
@@ -336,16 +366,137 @@ def scan(
                     if len(entries) == max_entries:
                         return entries
     except socket.error:
-        logging.getLogger(__name__).exception(
-            "Socket error while discovering SSDP devices"
-        )
+        LOG.exception("Socket error while discovering SSDP devices")
     finally:
         for s in sockets:
             s.close()
 
     return entries
 
 
+class DiscoveryResponder:
+    """Inform Wemo devices of the pywemo virtual Wemo device.
+
+    The DiscoveryResponder informs Wemo devices of the /setup.xml URL for the
+    pywemo virtual Wemo device. The virtual device is used for receiving long
+    press actions from Wemo devices and is integrated into the
+    SubscriptionRegistry HTTP server.
+
+    Wemo devices are informed of the pywemo virtual Wemo device in two ways:
+
+    1. Wemo devices periodically send UPnP M-SEARCH discovery requests for the
+    to locate other devices on the network. DiscoveryResponder responds to these
+    requests with the URL for the virtual device.
+
+    2. A UPnP NOTIFY message is periodically multicasted by DiscoveryResponder
+    to inform Wemo devices on the network of the URL for the virtual device.
+    """
+
+    def __init__(self, callback_port: int):
+        """Create a server that will respond to WeMo discovery requests.
+
+        Args:
+            callback_port: The port for the SubscriptionRegistry HTTP server.
+        """
+        self.callback_port = callback_port
+        self._thread = None
+        self._exit = threading.Event()
+        self._thread_exception = None
+        self._notify_enabled = True  # For tests.
```

Comment:
What is this comment here for, `For tests.`?  Is the notify only needed for test code?

Reply:
Ah, no. Maybe there is a better way to rephrase that. It's so that I can _disable_ the notify for tests. Outside of the test code, nothing else sets this to `False`. It's easier to test all the socket calls if notify can be tested independently from handling M-SEARCH. I added the comment so someone doesn't later come and look at the code and say "this is never set to False, why is it needed?"

Reply:
makes more sense, thanks